jupyterhub/roles.py

```diff
@@ -380,32 +385,36 @@ def _token_allowed_role(db, token, role):
     """Returns True if token allowed to have requested role through
     comparing the requested scopes with the set of token's owner scopes"""
 
-    standard_permissions = {'all', 'read:all'}
+    owner = token.user
+    if owner is None:
+        owner = token.service
```

Comment:
I love this, thanks! I thought we had to query db to find out who's the token owner.